src/reflect/scala/reflect/internal/BaseTypeSeqs.scala

```diff
@@ -130,6 +131,17 @@ trait BaseTypeSeqs {
     /** Return all evaluated types in this sequence as a list */
     def toList: List[Type] = elems.toList
 
+    def toIterator: Iterator[Type] = new AbstractIterator[Type] {
```

Comment:
Does anybody know if this kind of simple `Iterator` over an `Array` is already defined in the standard collection libary? I could not find it anywhere, and it may come handy for others.

Reply:
In 2.13 it exists as the private class `ArrayOps.ArrayIterator`.

Reply:
I've pushed a simplification to just use `elems.iterator`

Reply:
What does the `elems.iterator` do? Where is I defined? Is it by any chance creating a linked sequence and then iterating over it?

Reply:
```
scala> (as: Array[String]) => as.iterator //print<TAB>

((as: scala.Array[scala.Predef.String]) => scala.Predef.refArrayOps[String](as).iterator) // : Array[String] => Iterator[String]
```

It implicit converts `Array` to `ArrayOps`, then calls `ArrayOps.iterator` which is basically what you implemented.